src/modules/Dashboard/components/ActivityUnassignDrawer/ActivityUnassignDrawer.utils.tsx

```diff
@@ -0,0 +1,62 @@
+import { Trans } from 'react-i18next';
+import { t } from 'i18next';
+
+import { RespondentDetails } from 'modules/Dashboard/types';
+
+import { GetConfirmationBodyProps } from './ActivityUnassignDrawer.types';
+
+const getSubjectLabel = (subject?: RespondentDetails) => {
+  if (!subject) {
+    return '';
+  }
+
+  return subject.tag === 'Team'
+    ? `${subject.nickname}`
+    : `${subject.secretUserId}, ${subject.nickname}`;
+};
+
+export const getConfirmationBody = ({ selected, participantContext }: GetConfirmationBodyProps) => {
+  const respondent = getSubjectLabel(selected[0]?.respondentSubject);
+  const target = getSubjectLabel(selected[0]?.targetSubject);
+
+  if (selected.length === 1) {
+    return (
+      <Trans i18nKey="activityUnassign.confirmationBodySingle">
+        <strong>
+          <>{{ respondent }}</>
+        </strong>
+        <strong>
+          <>
+            {{
+              target:
+                selected[0]?.respondentSubject.id === selected[0]?.targetSubject.id
+                  ? t('activityUnassign.subjectSelf')
+                  : target,
+            }}
+          </>
+        </strong>
+      </Trans>
+    );
+  }
+
+  return participantContext === 'respondent' ? (
+    <Trans i18nKey="activityUnassign.confirmationBodyMultipleTargets">
+      <strong>
+        <>{{ respondent }}</>
+      </strong>
+      <strong>
+        <>{{ selectedCount: selected.length }}</>
+      </strong>
+    </Trans>
+  ) : (
+    <Trans i18nKey="activityUnassign.confirmationBodyMultipleRespondents">
+      <strong>
+        <>{{ selectedCount: selected.length }}</>
+      </strong>
+      <strong></strong>
+      <strong>
+        <>{{ target }}</>
+      </strong>
+    </Trans>
+  );
```

Comment:
Good eye, Kenroy.

**Re first comment:**

I should have called this out in the PR description. Here's the situation:

It's only in the context of this now-outdated PDP design that it's possible to initiate Unassign where there may be a combination of multiple respondents _and_ multiple target subjects in the list of assignments. It will not be possible in the new designs, because each "Unassign" trigger is from the context of a subject as the respondent, _or_ the subject as the target subject – no longer both.

However, now that you point it out, I think it's prudent to be flexible anyway, even if won't be possible with my upcoming PDP revisions. For example, maybe eventually (and perhaps soon even) we'll be able to perform "Unassign" outside of the context of a given subject (say, for an activity or flow in general), in which case it should be able to handle this case.

**Re second comment:**

I think the reason it's doing this is because `respondent` is hardcoded for the `participantContext` prop, so currently, it's always showing the wrong copy when there are multiple subjects in this temporary version of the PDP. I'll make more resilient so the copy actually reflects the list of assignments, it'll be a simple enough fix.